### PR TITLE
Adapt to the latest HPX API.

### PR DESCRIPTION
**Pull request by rtohid**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+## Build
+```sh
+cmake -S . -DHPX_DIR=/path/to/hpx/lib/cmake/HPX -B cmake-build/
+cmake --build cmake-build/ --parallel
+```
+
+> **NOTE:**
+>
+>HPX must be built with `HPX_WITH_DYNAMIC_HPX_MAIN=OFF`.
```
Comment (rtohid) [inline README.md:16]: I needed this to build the examples, not sure if it's needed in general.

Comment (hkaiser) [reply README.md:16]: Do the examples still run if you disable this option?

Comment (rtohid) [reply README.md:16]: It won't compile if it's not set to 'OFF'. I get `undefined reference to 'main'`.

With `HPX_WITH_DYNAMIC_HPX_MAIN=OFF`, examples run, but not all successfully. The 'counter' examples hang, and the cancel example not work as expected, printing "something went wrong."

Comment (hkaiser) [reply README.md:16]: Ok, thanks! If some of the examples don't work, then it's probably unrelated to this cmake option.

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Pull request merged**